master/internal/user/service.go

```diff
@@ -223,7 +223,12 @@ func (s *Service) postLogin(c echo.Context) (interface{}, error) {
 	// Get the user from the database.
 	user, err := s.db.UserByUsername(params.Username)
 	if err != nil {
-		return nil, badCredentialsError
+		switch err.(type) {
+		case db.ErrNoSuchUsername:
+			return nil, badCredentialsError
```

Comment:
non-blocking: may be worth throwing some warning logs around the badCredentialsErrors that we have here